### PR TITLE
Refactor yarpcjson to match new errors API

### PR DESCRIPTION
**Pull request by vyshah**

### CODE REVIEW

File: v2/yarpcjson/inbound.go [hunk 1]
```diff
@@ -42,7 +43,11 @@ type jsonHandler struct {
 func (h jsonHandler) Handle(ctx context.Context, reqBody interface{}) (interface{}, error) {
 	results := h.handler.Call([]reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(reqBody)})
 	if appErr, _ := results[1].Interface().(error); appErr != nil {
-		return results[0].Interface(), appErr
+		return results[0].Interface(), yarpcerror.New(
+			yarpcerror.CodeUnknown,
+			appErr.Error(),
+			yarpcerror.WithDetails(appErr),
```
Comment (kriskowal) [inline v2/yarpcjson/inbound.go]: When the inbound codec has `EncodeError(appError)`, do we expect it to figure out whether `appErr` is serializable as JSON, or should we make that distinction here?

This is cool.

Comment (vyshah) [reply v2/yarpcjson/inbound.go]: @robbertvanginkel

Comment (mh-park) [reply v2/yarpcjson/inbound.go]: The former, I think. The codec should handle all the serialization logic.

Comment (robbertvanginkel) [reply v2/yarpcjson/inbound.go]: The `appErr` shouldn't be in the details here, details should be empty if the user didn't supply a yarpcerror.

File: v2/yarpcjson/inbound.go [hunk 2]
```diff
@@ -42,7 +43,15 @@ type jsonHandler struct {
 func (h jsonHandler) Handle(ctx context.Context, reqBody interface{}) (interface{}, error) {
 	results := h.handler.Call([]reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(reqBody)})
 	if appErr, _ := results[1].Interface().(error); appErr != nil {
-		return results[0].Interface(), appErr
+		if yarpcerror.IsStatus(appErr) {
+			return results[0].Interface(), appErr
+		}
+
+		return results[0].Interface(), yarpcerror.New(
```
Comment (AlexAPB) [inline v2/yarpcjson/inbound.go]: Can this first return value be nil?

**Pull request closed without merging**